lua/neogit/popups/init.lua

```diff
@@ -0,0 +1,109 @@
+local M = {}
+
+---@param name string
+---@param f nil|fun(create: fun(...)): any
+--- Creates a curried function which will open the popup with the given name when called
+function M.open(name, f)
+  f = f or function(f)
```

Comment:
Can you explain how this works and why? 😅

Reply:
Yes absolutely.

It is a function which given a popup name will return a function (which takes 0 arguments) which will when called load the popup `neogit.popups/{name}` module, and then pass its `create` function to the second argument `f`.

`f` can then call the `create` function with any arguments it likes, or whenever it wants.

I originally had it be plain argument, and `create` was called inside the `open` returned function instead, but this allows async and sleeping, as well as aborting the popup using the supplied function.

Essentially, this function will create a keymapping ready function, which will supply `f` with the means to open the correct popup.

the `f = f or function(create) create() end` will simply call the `create` function directly with no arguments if no second argument is supplied. Most popups don't need any extra arguments, but some do, such as stashing.

The normal flow is 

`open` loads popup module
uses `f` to help calling the `popup.create` method.

In most cases, if the popup.create requires no special loading or arguments,

`popup.open("my_awesome_popup") -> function() which opens popup` is enough

This is usually called *lifting* and creating a function which "fills in" some arguments and returns a function, eg `function(a,b) -> function(b)` is called currying (partial application)